### PR TITLE
kops: Bump terraform providers and modules for the build cluster.

### PR DESCRIPTION
**Pull request by ameukam**

### CODE REVIEW

**Comment by ameukam** [comment]
cc @bryantbiggs

File: infra/aws/terraform/kops-infra-ci/eks.tf [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 module "eks" {
   providers = { aws = aws.kops-infra-ci }
   source    = "terraform-aws-modules/eks/aws"
-  version   = "20.8.3"
+  version   = "20.10.0"
```
Comment (bryantbiggs) [inline infra/aws/terraform/kops-infra-ci/eks.tf:20]: any reason not to go up to `2.20.0`?

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/eks.tf:20]: I don't think we need the latest version but happy to do it.

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/eks.tf:20]: For this cluster, there is no real need to switch to Karpenter right now. Probably something to think about in 2025.

**Review by upodroid — APPROVED** [review]

**Comment by ameukam** [comment]
@bryantbiggs I got

```bash
│ Error: updating EKS Node Group (k8s-infra-kops-prow-build:prow-build-20231205235350132400000001) version: operation error EKS: UpdateNodegroupVersion, https response error StatusCode: 400, RequestID: cc844a55-f471-43ab-9900-16b33e4faa3a, InvalidRequestException: 'k8s.io/cluster-autoscaler/k8s-infra-kops-prow-build' is not a valid tag key. Tag keys must match pattern ([0-9a-zA-Z\\-_+=,.@:]{1,255}), and must not be a reserved name ('.', '..', '_index')
``` 

but I'm sure the tags didn't change.

**Comment by bryantbiggs** [comment]
it might be new validation that MNG is doing - we shouldn't need to add [these tags](https://github.com/kubernetes/k8s.io/pull/7036/files#diff-e796ca0bc77afd702381aaf5c872e6017375f53adc25c951c5689a510638413bR132-R133) per https://github.com/aws/containers-roadmap/issues/608#issuecomment-639549332

**Pull request merged**